drivers/i2c/Kconfig.gd32

```
@@ -0,0 +1,10 @@
+# Copyright (c) 2021 BrainCo Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_GD32_I2C := gd,gd32-i2c
+
+config I2C_GD32
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_GD32_I2C))
```

I think this should be left as it was:

```suggestion
	bool "GigaDevice GD32 I2C driver"
	default $(dt_compat_enabled,$(DT_COMPAT_GD32_I2C))
```

ref. https://discord.com/channels/720317445772017664/733037557218279515/930027026994331709

---

So, we should give the final user a chance to disable it.

---

done